File: src/gestorAplicacion/personas/Medico.java
package gestorAplicacion.personas;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.Map;
import java.util.TreeMap;
import gestorAplicacion.registrosMedicos.*;
//import gestorAplicacion.Cita;
import gestorAplicacion.Consulta;
import gestorAplicacion.Consultorio;
import gestorAplicacion.Medicamentos;
import gestorAplicacion.Pago;
import gestorAplicacion.tipoCita;

public class Medico extends Persona{
	private static final long serialVersionUID = 1L;
    public static ArrayList<Diagnostico> listado = new ArrayList<Diagnostico>();
    public tipoCita especialidad;
    public tipoMedico especialista;
    private boolean disponibilidad;
    public Map<LocalDateTime, Consulta> consultas = new TreeMap<LocalDateTime, Consulta>();
    public static ArrayList<Medico> medicos = new ArrayList<Medico>();
    public static ArrayList<Consulta> listado_consulta = new ArrayList<Consulta>(); //crea la consulta
    public Map<LocalDateTime, Pago> nomina = new TreeMap<LocalDateTime, Pago>();
    private long sueldo;
    public  ArrayList<Medicamentos> medicamiento = new ArrayList<Medicamentos>();

    
    //constructor

    public Medico(String nombre, String apellido, tipoDocumento documento, long numeroDocumento, int edad,
            String genero, long telefono, String correoElectronico, String direccion, tipoCita especialidad,
            tipoMedico especialista, boolean disponibilidad, Map<LocalDateTime, Consulta> consultas,
            Map<LocalDateTime, Pago> nomina, long sueldo, ArrayList<Medicamentos> medicamiento) {
        super(nombre, apellido, documento, numeroDocumento, edad, genero, telefono, correoElectronico, direccion);
        this.especialidad = especialidad;
        this.especialista = especialista;
        this.disponibilidad = disponibilidad;
        this.consultas = consultas;
        medicos.add(this);
        this.nomina = nomina;
        this.sueldo = sueldo;
        this.medicamiento = medicamiento;

    }
   
    public Medico(tipoCita especialidad, tipoMedico especialista, boolean disponibilidad,
            Map<LocalDateTime, Consulta> consultas) {
        this.especialidad = especialidad;
        this.especialista = especialista;
        this.disponibilidad = disponibilidad;
        this.consultas = consultas;
        medicos.add(this);
    }

    public Medico(String nombre, String apellido, tipoDocumento documento, int numeroDocumento, int edad, String genero, int telefono,
			String correoElectronico, String direccion, tipoCita especialidad, tipoMedico especialista, boolean disponibilidad) {
    	super(nombre, apellido, documento, numeroDocumento, edad, genero, telefono, correoElectronico, direccion);
    	this.especialidad = especialidad;
        this.especialista = especialista;
        this.disponibilidad = disponibilidad;
        medicos.add(this);
    	
    }
    	

    public Medico(tipoCita especialidad, tipoMedico especialista, boolean disponibilidad,
            Map<LocalDateTime, Consulta> consultas, Map<LocalDateTime, Pago> nomina, long sueldo,
            ArrayList<Medicamentos> medicamiento) {
        this.especialidad = especialidad;
        this.especialista = especialista;
        this.disponibilidad = disponibilidad;
        this.consultas = consultas;
        this.nomina = nomina;
        this.sueldo = sueldo;
        this.medicamiento = medicamiento;
        medicos.add(this);
    }

    public Medico(){super();};

    
		// TODO Auto-generated constructor stub
	

	public Medico(String nombre, String apellido, tipoDocumento documento, long numeroDocumento , int edad, String genero, long telefono,
			String correoElectronico, String direccion, tipoCita especialidad, boolean disponibilidad) {
		 super(nombre, apellido, documento, numeroDocumento, edad, genero, telefono, correoElectronico, direccion);
	        this.especialidad = especialidad;
	        this.disponibilidad = disponibilidad;
	        medicos.add(this);
	}

	//getter & setter
    public long getSueldo() {
        return sueldo;
    }
    public void setSueldo(long sueldo) {
        this.sueldo = sueldo;
    }

    public Map<LocalDateTime, Pago> getNomina() {
        return nomina;
    }

    public void setNomina(Map<LocalDateTime, Pago> nomina) {
        this.nomina = nomina;
    }

    public tipoCita getEspecialidad() {
        return especialidad;
    }

    public void setEspecialidad(tipoCita especialidad) {
        this.especialidad = especialidad;
    }

    public boolean isDisponibilidad() {
        return disponibilidad;
    }

    public void setDisponibilidad(boolean disponibilidad) {
        this.disponibilidad = disponibilidad;
    }

    public Map<LocalDateTime, Consulta> getConsultas() {
        return consultas;
    }

    public void setConsultas(Map<LocalDateTime, Consulta> consultas) {
        this.consultas = consultas;
    }

    public tipoMedico getEspecialista() {
        return especialista;
    }

    public void setEspecialista(tipoMedico especialista) {
        this.especialista = especialista;
    }
    
    
    
    //metodos
    public void crearDiagnostico(Paciente persona, ArrayList<String> sintomas, String observacion, Medicamentos medicamiento, int id, Consulta consulta){
        Diagnostico diag = new Diagnostico (persona, sintomas, observacion, medicamiento, id, consulta);
        listado.add(diag);
    }
    public static int informesMedicos(tipoMedico solicitado) {// se ingresa el tipo de examen y se busca en la lista cuantos hay
        int tmSolicitado = 0;
        for (Medico tm : medicos) {
            if (tm.getEspecialista() == solicitado) {
                tmSolicitado++;
            }
        }
        return tmSolicitado;
    }


}

File: Python/src/gestorAplicacion/personas/Medico.py
from gestorAplicacion.personas.Persona import Persona
from gestorAplicacion.personas.Administrador import Administrador
from gestorAplicacion.registrosMedicos import Diagnostico
class Medico(Persona):
    def __init__(self, nombre, apellido, documento, numeroDocumento, edad, genero, telefono, correoElectronico,
                 direccion, especialidad, especialista, contratado = False, disponibilidad = False, consultas = None, 
                 nomina = None, sueldo = 50000, medicamento = None):
        super().__init__(nombre, apellido, documento, numeroDocumento, edad, genero, telefono, correoElectronico, 
                        direccion)
        if consultas is None:
            consultas = {}
        if nomina is None:
            nomina = {}
        if medicamento is None:
            medicamento = []
        self._especialidad = especialidad
        self._especialista = especialista
        self._disponibilidad = disponibilidad
        self._consultas = consultas
        Administrador.medicos.append(self)
        self.nomina = nomina
        self.sueldo = sueldo
        self.medicamiento = medicamento
        self._contratado = contratado
        

    # Getters y Setters
    @classmethod
    def getSueldo(self):
        return self._sueldo
    
    def setSueldo(self, sueldo):
        self._sueldo = sueldo
    
    def getNomina(self):
        return self._nomina
    
    def setNomina(self, nomina):
        self._nomina = nomina

    def getEspecialidad(self):
        return self._especialidad
    
    def setEspecialidad(self, especialidad):
        self._especialidad = especialidad

    def isDisponibilidad(self):
        return self._disponibilidad
    
    def setDisponibilidad(self, disponibilidad):
        self._disponibilidad = disponibilidad

    def getConsultas(self):
        return self._consultas
    
    def setConsultas(self, consultas):
        self._consultas = consultas

    def getEspecialista(self):
        return self._especialista
    
    def setEspecialista(self, especialista):
        self._especialista = especialista

    def isContratado(self):
        return self._contratado
    
    def setContratado(self, contratado):
        self._contratado = contratado

    listado=[]
    @staticmethod
    def crearDiagnostico(persona, sintomas, observacion, medicamiento, consulta):
        from gestorAplicacion.personas.Medico import Medico
        diag = Diagnostico (persona, sintomas, observacion, medicamiento, consulta)
        Medico.listado.append(diag)
